src/classes/CON_Email.cls

```diff
@@ -0,0 +1,132 @@
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+/**
+* @author Salesforce.org
+* @date 2016
+* @group Contacts
+* @group-content ../../ApexDocContent/Contacts.htm
+* @description A Class to handle Contact Email related Items
+*
+*/
+public class CON_Email {
+
+    /*******************************************************************************************************
+    * @description Sets the Email field depending on the value of Preferred Email
+    * @param Contact the contact to update
+    */
+	public static void processPreferredEmail(Contact contact, List<Contact> oldlist) {
+
+		Boolean isUpdate =  (oldlist != null) ? true : false;
+
+	    // Store default email field mappings to preferred email pick list.
+	    Map<String,String> preferredEmailMappings = new Map<String,String>{   
+	        'University' => 'UniversityEmail__c',
+	        'Work' => 'WorkEmail__c',
+	        'Alternate' => 'AlternateEmail__c'
+	    };
+
+	    Map<String,String> valuedEmailMap = emailFieldsWithValue(contact);
+
+        if( valuedEmailMap.size() > 0 ) {
+
+            // Enforce preferred email field
+            if(String.isBlank(contact.Preferred_Email__c)){
+                contact.addError( Label.PreferredEmailRequiredError );
+            }
+
+            // Get the Api name that matches the value in Preferred Email
+            String preferredApiField = ( preferredEmailMappings.containsKey(contact.Preferred_Email__c) ) ? preferredEmailMappings.get(contact.Preferred_Email__c) : valuedEmailMap.get(contact.Preferred_Email__c);
```

Comment:
Oh, I think I understand how. We first look at our custom email fields. If the preferred isn't one of them, we look at any other field of Email type, correct?

Maybe the easiest to make sure the dropdown values are translatable would be to avoid the mapping altogether?